command_test.go

```diff
@@ -3204,16 +3204,55 @@ func TestCommand_Bool(t *testing.T) {
 }
 
 func TestCommand_Value(t *testing.T) {
-	set := flag.NewFlagSet("test", 0)
-	set.Int("myflag", 12, "doc")
-	parentSet := flag.NewFlagSet("test", 0)
-	parentSet.Int("top-flag", 13, "doc")
-	pCmd := &Command{flagSet: parentSet}
-	cmd := &Command{flagSet: set, parent: pCmd}
+	subCmd := &Command{
+		Name: "test",
+		Flags: []Flag{
+			&IntFlag{
+				Name:    "myflag",
+				Usage:   "doc",
+				Aliases: []string{"m", "mf"},
+			},
+		},
+		Action: func(ctx context.Context, c *Command) error {
+			return nil
+		},
+	}
+
+	cmd := &Command{
+		Flags: []Flag{
+			&IntFlag{
+				Name:    "top-flag",
+				Usage:   "doc",
+				Aliases: []string{"t", "tf"},
+			},
+		},
+		Commands: []*Command{
+			subCmd,
+		},
+	}
+	err := cmd.Run(buildTestContext(t), []string{"main", "--top-flag", "13", "test", "--myflag", "14"})
+	assert.NoError(t, err)
 
 	r := require.New(t)
-	r.Equal(12, cmd.Value("myflag"))
-	r.Equal(13, cmd.Value("top-flag"))
+	r.Equal(int64(13), cmd.Value("top-flag"))
+	r.Equal(int64(13), cmd.Value("t"))
+	r.Equal(int64(13), cmd.Value("tf"))
+
+	r.Equal(int64(14), subCmd.Value("myflag"))
+	r.Equal(int64(14), subCmd.Value("m"))
+	r.Equal(int64(14), subCmd.Value("mf"))
+
+	// use only aliases here
+	err = cmd.Run(buildTestContext(t), []string{"main", "-tf", "15", "test", "-m", "16"})
+	assert.NoError(t, err)
+
+	r.Equal(int64(15), cmd.Value("top-flag"))
+	r.Equal(int64(15), cmd.Value("t"))
+	r.Equal(int64(15), cmd.Value("tf"))
+
+	r.Equal(int64(16), subCmd.Value("myflag"))
+	r.Equal(int64(16), subCmd.Value("m"))
+	r.Equal(int64(16), subCmd.Value("mf"))
```

Comment:
Just a small suggestion. I think we can make the test more self-documenting by using subtests.

```go
    t.Run("flag name", func(t *testing.T) {
	err := cmd.Run(buildTestContext(t), []string{"main", "--top-flag", "13", "test", "--myflag", "14"})
	assert.NoError(t, err)
	r := require.New(t)
	...
    })

    t.Run("flag alias", func(t *testing.T) {
	err := cmd.Run(buildTestContext(t), []string{"main", "-tf", "15", "test", "-m", "16"})
	assert.NoError(t, err)
	r := require.New(t)
	...
    })
```

Reply:
@Juneezee Thats an excellent idea. I've made the change